### PR TITLE
fix: minted count

### LINKED ISSUE #10901: Art Cards mint count
Please check all drops by yourself i just had a look at some but the numbers really should match!

grafik
grafik

grafik
grafik

grafik
grafik

grafik
grafik

grafik
grafik
grafik

grafik
grafik

### LINKED ISSUE #10904: Mint count doesnt work // is too slow to load
### What happened?

.

### Please reproduce in steps

.

### Expected Behavior

.

### What browsers are you seeing the problem on?

Chrome

### At which address did you encounter bug?

kodadot.xyz

### Are you logged in?

None

### Which wallet you are using?

_No response_

### Screenshots

Screenshot 2024-08-26 at 9 20 28

### Wallet Address & Transaction hash

_No response_

### Relevant log output

_No response_

### Payment link for reward

https://canary.kodadot.xyz/dot/transfer/?target=<My_Polkadot_Address_check_https://github.com/kodadot/nft-gallery/blob/main/REWARDS.md#creating-your-dot-address>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #10914: Circuits drop isn’t loading at all
image

- production - kodadot.xyz
- chrome
- MNT

https://kodadot.xyz/mnt/drops/circuits

### LINKED ISSUE #10916: Base: Error while minitng Quadz
Collection: https://kodadot.xyz/base/drops/quadz its a free collection on Base

I know its testing collection but i got an ugly error while trying to mint it

Screenshot 2024-08-29 at 13 50 45

Video:

https://github.com/user-attachments/assets/a4c02129-397b-402c-9fbe-9d2b33ecf095

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #10901
- [x] Closes #10904
- [x] Closes #10914
- [x] Closes #10916
- [x] Requires worker deployment https://github.com/kodadot/private-workers/pull/213

tldr; https://github.com/kodadot/nft-gallery/pull/10903#issuecomment-2337376568

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)

### CODE REVIEW

**Comment by Jarsen136** [comment]
It may not be related to this issue.

After I minted a drop, the number of `minted` does not update. I have to refresh the page to see the correct number.

image

**Comment by preschian** [comment]
> It may not be related to this issue.
> 
> After I minted a drop, the number of `minted` does not update. I have to refresh the page to see the correct number.

this is really good point and related to this PR
hopefully it can clear the issue if we add popup message like this. let me ask Exec first (the popup is from fxhash)

E62CB9F8-22F0-4C48-A10C-D78BF7BA8A35_1_105_c

File: components/collection/drop/GenerativeLayout.vue [hunk 1]
```diff
@@ -89,16 +89,9 @@
...
-  Drop } from '@/components/drops/useDrops'
-import {
-  getFormattedDropItem,
-} from '@/components/drops/useDrops'
-import { useCollectionActivity } from '@/composables/collectionActivity/useCollectionActivity'
-import { useCollectionMinimal } from '@/components/collection/utils/useCollectionDetails'
 import useCursorDropEvents from '@/composables/party/useCursorDropEvents'
 import { DropEventType } from '@/composables/party/types'
-import type { DropItem } from '@/params/types'
+import { fetchOdaCollectionOwners } from '~/services/oda'
```
Comment (hassnian) [inline components/collection/drop/GenerativeLayout.vue:94]:
```suggestion
import { fetchOdaCollectionOwners } from '@/services/oda'
```

File: components/collection/drop/Phase.vue [hunk 2]
```diff
@@ -101,21 +101,6 @@
...
-    >
-      <p class="capitalize">
-        Location Verification: You are
-        <span
-          v-if="!Boolean(drop?.userAccess)"
-          class="font-bold"
-        >not</span> in
-        <span class="font-bold">{{ drop?.location }}</span>
-      </p>
-    </CollectionDropRequirementItem>
```
Comment (hassnian) [inline components/collection/drop/Phase.vue:118]: why are we removing this ?

File: components/drops/BasicDropCard.vue [hunk 3]
```diff
@@ -134,11 +134,11 @@ const props = withDefaults(
 const { placeholder } = useTheme()
 const isUnlimited = computed(() => props.dropMax > Number.MAX_SAFE_INTEGER)
 
-const chainPropList = chainPropListOf(props.dropPrefix)
+const chainPropList = chainPropListOf(props.dropPrefix ?? 'ahp')
```
Comment (hassnian) [inline components/drops/BasicDropCard.vue:137]: doesn't the prop already have this default set ? , if you still need it here make it a constant and also use it inside deineProps 

```suggestion
const chainPropList = chainPropListOf(props.dropPrefix ?? DEFAULT_DROP_PREFIX)
```

File: components/drops/DropItem.vue [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+const props = defineProps<{
+  drop: DropItem
+}>()
+
+const formattedDrop = ref<DropItem>()
+const loading = ref(true)
+const pastItem = computed(() => !formattedDrop.value?.isMintedOut)
+
+onBeforeMount(async () => {
+  formattedDrop.value = await getDropAttributes(props.drop.alias)
```
Comment (hassnian) [inline components/drops/DropItem.vue:25]: isn't this blocking? can we have it show the skeleton if `!formattedDrop`

**Review by hassnian — APPROVED** [review]

File: components/drops/BasicDropCard.vue (hunk 3 above)
Comment (preschian) [reply components/drops/BasicDropCard.vue]: oh correct, I removed `null` in the props instead `dropPrefix?: Prefix | null`

File: components/drops/DropItem.vue (hunk 4 above)
Comment (preschian) [reply components/drops/DropItem.vue:25]: Should be not blocking, use loading state in there. But, I adjusted it to be able to show minted drops. And it's using your suggestion `!formattedDrop`.

**Comment by preschian** [comment]
### Changes:

- Adjust carousel drops to align with [ops-internal discussion #1845](https://github.com/kodadot/ops-internal/discussions/1845)
- Remove `useDrops()` composable. Use `getDrops()` directly.
- Replace `useCollectionActivity()` usage with `oda-worker` to get owner address on `DropCard` related components:
  - Reason: `useCollectionActivity()` composable is computationally heavy on the client side.
  - Impact: When switching chains, it was noticeably laggy on localhost.
- Add toast message when claiming drops is successful:
  - Purpose: To provide more information to users about the delay in seeing the drops on the website.
  - Reference: [PR #10903 comment](https://github.com/kodadot/nft-gallery/pull/10903#issuecomment-2324043010)
- Remove `drop.location` usage:
  - Rationale: Seems to be rarely used. Can be added back if needed in the future.
- Refactor some `useDrops()` composable to `drops/utils.ts`
- Remove `Drop` interface, adjust `DropItem` instead
- Remove unused `Unlockable` components

**Review by hassnian — APPROVED** [review]

**Comment by vikiival** [comment]
cc @Jarsen136

File: stores/dropsMinted.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  state: () => ({
+    mintedDrops: [] as DropItem[],
+  }),
+  getters: {
+    sortedMintedDrops(): DropItem[] {
+      return [...this.mintedDrops].sort((a, b) => {
+        // Sort from bigger to lower number
+        return Number(b.collection) - Number(a.collection)
+      })
+    },
```
Comment (Jarsen136) [inline stores/dropsMinted.ts:16]: For the drops on base, the id of collection is not the number but the address.

image

**Review by Jarsen136 — APPROVED** [review]
otherwise lgtm

File: stores/dropsMinted.ts (hunk 5 above)
Comment (vikiival) [reply stores/dropsMinted.ts:16]: > `For the drops on base, the id of collection is not the number but the address.`

Well in the end the address is just 20-byte integer ^-^ 

```js
> a = '0x8aedf51f1643296b83dc280680bad43237dfb52a'
'0x8aedf51f1643296b83dc280680bad43237dfb52a'
> b = '0x9789b13d485b124255f9dbf42aa7091aa22e87f8'
'0x9789b13d485b124255f9dbf42aa7091aa22e87f8'
> Number(a) - Number(b)
// -7.198089214047003e+46
```

Comment (Jarsen136) [reply stores/dropsMinted.ts:16]: That's true. However, I guess it is supposed to sort the collections by the collection creation time. In other words, sorting by collection ID can not achieve this goal for base drops.

Comment (preschian) [reply stores/dropsMinted.ts:19]: updated, let's show it without no sorted method for evm base. since we doesnt have `created_at` data

**Comment by vikiival** [comment]
remedy for - #10958
